daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -448,107 +479,110 @@ object SExpr {
       final case object EqualList extends Reference
     }
 
-    val FoldL: SEBuiltinRecursiveDefinition = SEBuiltinRecursiveDefinition(Reference.FoldL)
-    val FoldR: SEBuiltinRecursiveDefinition = SEBuiltinRecursiveDefinition(Reference.FoldR)
-    val EqualList: SEBuiltinRecursiveDefinition = SEBuiltinRecursiveDefinition(Reference.EqualList)
-
-    private val foldLBody: SExpr =
-      // foldl f z xs =
-      SEMakeClo(
-        Array(),
-        3,
-        // case xs of
-        SECase(SELocA(2)) of (
-          // nil -> z
-          SCaseAlt(SCPNil, SELocA(1)),
-          // cons y ys ->
-          SCaseAlt(
+    val FoldL = SEBuiltinRecursiveDefinition(Reference.FoldL)
+    val FoldR = SEBuiltinRecursiveDefinition(Reference.FoldR)
+    val EqualList = SEBuiltinRecursiveDefinition(Reference.EqualList)
+
+    // The body of an expanded recursive-builtin will always be in ANF form.
+    // The comments showing where variables are to be found at runtime have been
+    // fully checked and updated during the move to ANF.
```

Comment:
Does it make sense to have a comment saying the comments below are up-to-date? It's nice for me as a review to know but all futures readers of the code will assume the comments are up-to-date anyway.

Reply:
ok, fair enough

Reply:
fixed